### PR TITLE
Mutate utils part deux

### PR DESCRIPTION
**Pull request by pablisco**
- Organises utils into separate files
- Adds operators to remove items by returning a null instead of a value.
- Adds operators to remove items that don't match a given type.
- Makes use we are using iterators as much as possible. Otherwise, we are doing a look-up for each mutation. Haven't found a way to modify an item inside a `MutableSet` so we are collecting the non-null values.
- Adds tests for these utils.

### CODE REVIEW

File: mutable-utils/src/test/kotlin/at/kopyk/TestMutation.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package at.kopyk
+
+import io.kotest.matchers.types.shouldBeSameInstanceAs
+
+internal fun <T> testMutation(
```
Comment (serras) [inline mutable-utils/src/test/kotlin/at/kopyk/TestMutation.kt:5]: Loving this combinator!

Comment (pablisco) [reply mutable-utils/src/test/kotlin/at/kopyk/TestMutation.kt:5]: Thank you 🥰
I love to create things like these as well as test scopes for more complex cases. Beats doing shared setup and teardown strategies in so many ways hehe.

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -15,6 +16,7 @@ kotlinPoet-ksp = { module = "com.squareup:kotlinpoet-ksp", version.ref = "kotlin
 ksp = { module = "com.google.devtools.ksp:symbol-processing-api", version.ref = "kspVersion" }
 kotlinCompileTesting = { module = "com.github.tschuchortdev:kotlin-compile-testing", version.ref = "kotlinCompileTesting" }
 kotlinCompileTestingKsp = { module = "com.github.tschuchortdev:kotlin-compile-testing-ksp", version.ref = "kotlinCompileTesting" }
+kotest-assertions-core = { module = "io.kotest:kotest-assertions-core", version.ref = "kotest" }
 assertj = { module = "org.assertj:assertj-core", version.ref = "assertj" }
```
Comment (serras) [inline gradle/libs.versions.toml:20]: Do you think we can somehow use only Kotest and remove AssertJ? 🤔 

(Not for this PR, of course)

**Review by serras — APPROVED** [review]
Nice job! I think the set of combinators is a good choice, and should lead to nice code.

Just missing... a few changes in the README to point to these new utilities. I'll try to work on a PR which generates Dokka documentation from this.

File: gradle/libs.versions.toml (hunk 2 above)
Comment (pablisco) [reply gradle/libs.versions.toml:20]: Oh, yeah. I missed AssertJ there 🙈 I'll create a separate PR to convert to kotest (which is more kotlin friendly IMHO hehe)

**Pull request merged**